src/solvers/flattening/arrays.cpp

```diff
@@ -547,13 +511,8 @@ void arrayst::add_array_constraints_with(
   // use other array index applications for "else" case
   // add constraint x[I]=y[I] for I!=i
 
-  for(index_sett::const_iterator
-      it=index_set.begin();
-      it!=index_set.end();
-      it++)
+  for(auto other_index : index_set)
```

Comment:
That won't work, as other_index may be changed later on; i.e., the copy is intentional.